astroquery/desi/tests/test_desi.py

```diff
@@ -0,0 +1,118 @@
+import astropy.io.votable
+import pytest
+import os
+import numpy as np
+import pyvo as vo
+
+from astropy.table import Table
+from astropy.io import fits
+from astropy import coordinates as coord
+from astroquery.utils.mocks import MockResponse
+from astroquery.utils import commons
+from astroquery import desi
+from urllib import parse
+from contextlib import contextmanager
+
+
+DATA_FILES = {
+    'dummy_tap_table': 'dummy_table.txt',
+    'dummy_tractor_fits': 'dummy_tractor.fits',
+    'dummy_hdu_list_fits': 'hdu_list_images.fits'
+}
+
+coords = commons.ICRSCoord('11h04m27s +38d12m32s')
+radius = coord.Angle(0.5, unit='arcmin')
+pixels = 60
+data_release = 9
+emispheres_list = ['north', 'south']
+
+
+@pytest.fixture
+def patch_get(request):
+    mp = request.getfixturevalue("monkeypatch")
+
+    mp.setattr(desi.DESILegacySurvey, '_request', get_mockreturn)
+    return mp
+
+
+@pytest.fixture
+def patch_get_readable_fileobj(request):
+    @contextmanager
+    def get_readable_fileobj_mockreturn(filename, **kwargs):
+        file_obj = data_path(DATA_FILES['dummy_hdu_list_fits'])  # TODO: add images option
+        f = None
+        try:
+            f = open(file_obj, 'rb')
+            yield f
+        finally:
+            if f is not None:
+                f.close()
+
+    mp = request.getfixturevalue("monkeypatch")
+
+    mp.setattr(commons, 'get_readable_fileobj',
+               get_readable_fileobj_mockreturn)
+    return mp
+
+
+@pytest.fixture
+def patch_tap(request):
+    mp = request.getfixturevalue("monkeypatch")
+
+    mp.setattr(vo.dal.TAPService, 'run_sync', tap_mockreturn)
+    return mp
+
+
+def get_mockreturn(method, url, params=None, timeout=10, **kwargs):
+    parsed_url = parse.urlparse(url)
+    splitted_parsed_url = parsed_url.path.split('/')
+    url_filename = splitted_parsed_url[-1]
+    filename = None
+    content = None
+    if url_filename.startswith('tractor-'):
+        filename = data_path(DATA_FILES['dummy_tractor_fits'])
+
+    if filename is not None:
+        content = open(filename, 'rb').read()
```

Comment:
It's better to use `open()` together with the `with` keyword, see https://docs.python.org/3/tutorial/inputoutput.html#reading-and-writing-files
```suggestion
        with open(filename, 'rb') as f:
            content = f.read()
```